src/operator/tensor/sample_op.h

```diff
@@ -92,16 +109,64 @@ void SampleNormal_(const nnvm::NodeAttrs& attrs,
   using namespace mxnet::op;
   using namespace mshadow::expr;
   mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
-  CHECK_EQ(outputs[0].type_flag_, mshadow::kFloat32)
-      << "only support float32 rnd so far";
   const SampleNormalParam& param = nnvm::get<SampleNormalParam>(attrs.parsed);
   mshadow::Random<xpu, float> *prnd = ctx.requested[0].get_random<xpu, float>(s);
-  mshadow::Tensor<xpu, 2, float> out = outputs[0].FlatTo2D<xpu, float>(s);
-  prnd->SampleGaussian(&out, param.loc, param.scale);  // NOLINT(*)
+  MSHADOW_REAL_TYPE_SWITCH(outputs[0].type_flag_, DType, {
+    if (outputs[0].type_flag_ != mshadow::kFloat32) {
+      // Not float32: use workspace and copy to output
```

Comment:
this line `MSHADOW_REAL_TYPE_SWITCH(outputs[0].type_flag_, DType, {` can be moved here inside the if right?